modules/product/src/main/java/com/opengamma/strata/product/fx/type/ImmutableFxSwapConvention.java

```diff
@@ -157,19 +160,23 @@ public FxSwapTrade toTrade(
       LocalDate startDate,
       LocalDate endDate,
       BuySell buySell,
+      Currency buySellCurrency,
       double notional,
       double nearFxRate,
       double farLegForwardPoints) {
 
+    if (!currencyPair.contains(buySellCurrency)) {
+      throw new IllegalArgFailureException("Buy/Sell currency {value} does not match convention {options}", buySellCurrency, currencyPair);
+    }
     Optional<LocalDate> tradeDate = tradeInfo.getTradeDate();
     if (tradeDate.isPresent()) {
       ArgChecker.inOrderOrEqual(tradeDate.get(), startDate, "tradeDate", "startDate");
     }
-    double amount1 = BuySell.BUY.normalize(notional);
+    double amount1 = buySell.normalize(notional);
```

Comment:
This is the key fix. It means the result takes account of the Buy/Sell flag where previously the flag was ignored.